node/src/main/scala/com/wavesplatform/api/http/eth/EthRpcRoute.scala

```diff
@@ -69,12 +68,14 @@ class EthRpcRoute(blockchain: Blockchain, transactionsApi: CommonTransactionsApi
         case "eth_getTransactionCount" =>
           resp(id, quantity(time.getTimestamp()))
         case "eth_getBlockByNumber" =>
-          resp(
-            id,
-            Json.obj(
-              "number" -> quantity(Integer.parseInt(param1Str.drop(2), 16))
-            )
-          )
+          val blockNumber = param1Str match {
+            case "earliest" => Some(1)
+            case "latest"   => Some(blockchain.height)
+            case "pending"  => None
+            case _          => Some(Integer.parseInt(param1Str.drop(2), 16))
+          }
+          val result = blockNumber.map(n => JsString(quantity(n))).getOrElse(JsNull)
```

Comment:
Do we still need `JsNull ` if block number is always returned as suggested above?

Reply:
Yes, according to the specification https://www.quicknode.com/docs/ethereum/eth_getBlockByNumber